indexer/SymbolFormatter.cc

```diff
@@ -220,6 +220,20 @@ SymbolFormatter::getEnumSymbol(const clang::EnumDecl *enumDecl) {
   return this->getTagSymbol(static_cast<const clang::TagDecl *>(enumDecl));
 }
 
+std::optional<std::string_view>
+SymbolFormatter::getNamedDeclSymbol(const clang::NamedDecl *namedDecl) {
+#define HANDLE(kind_)            \
```

Comment:
This is the first (and only) place where we're iterating over the separate kinds in a `NamedDecl`, so I figured it'd be simpler to use a first-order macro rather than a higher-order macro like in V8.

Reply:
You also declare all of the getters with the particular signature used here tho.

Reply:
So there are two different things here. The list of declarations in `Indexer.h` will eventually end up including most/all subclasses of Decl. The list of kinds here includes subclasses of NamedDecl. They're two different lists conceptually, even though right now they're the same.

I'm going to keep this as-is, because my code navigation experience in VS Code with similar declarations has been less than stellar, so I don't think it's worth de-duplicating that right now. We can revisit this later.

Reply:
The other thing is, I do not want to hide the list of declarations being handled, because having the list be directly visible in context is useful to quickly see "OK, these cases are not being handled explicitly". Adding an extra level of macro would add extra indirection for understanding that.

Once more switches are exhaustive, I think it may make sense to factor this out further, because it's clear that all cases are being handled.